doc/coq.rst

```diff
@@ -0,0 +1,383 @@
+
+We may indistinctively refer to a :ref:`dune-project` containing Coq theories as
+a *Coq project*.
+
+A *Coq project* is a collection of Coq plugins and Coq theories. This notion is
+subsumed by that of a :ref:`dune-project`, and should not be considered a
+technical definition.
+
+The ``.v`` files of a theory do not only have to be present as source files.
+They may also be created as Dune targets from other rules.
+
+.. _coq-theory:
+
+coq.theory
+----------
+
+To enable Coq support in a Dune project. The Coq language version should be
+selected in the :ref:`dune-project` file. For example:
+
+.. code:: scheme
+
+    (using coq 0.3)
+
+This will allow for the ``coq.theory`` stanza and other ``coq.*`` stanzas to be
+used in the current project. See the :ref:`Dune Coq language section<coq-lang>`
+for more details.
+
+The Coq theory stanza is very similar in form to the OCaml :ref:`library`
+stanza:
+
+.. code:: scheme
+
+    (coq.theory
+     (name <module_prefix>)
+     (package <package>)
+     (synopsis <text>)
+     (modules <ordered_set_lang>)
+     (libraries <ocaml_libraries>)
+     (flags <coq_flags>)
+     (mode <coq_native_mode>)
+     (theories <coq_theories>))
+
+The stanza will build all ``.v`` files in the given directory and also all
+subdirs if the :ref:`include-subdirs <include-subdirs-coq>` stanza is present.
+The semantics of fields are:
+
+- ``<module_prefix>`` is a dot-separated list of valid Coq module names and
+  determines the module scope under which the theory is compiled (this
+  corresponds to Coq's ``-R`` option).
+  
+  For example, if ``<module_prefix>`` is ``foo.Bar``, the theory modules will be
+  named as ``foo.Bar.module1``, ``foo.Bar.module2``, etc. Note that modules in
+  the same theory don't see the ``foo.Bar`` prefix in the same way that OCaml
+  ``wrapped`` libraries do.
+  
+  For compatibility reasons, Coq lang 1.0 will install a theory named
+  ``foo.Bar`` under ``foo/Bar``. Also note that Coq supports composing a module
+  path from different theories, thus you can name a theory ``foo.Bar`` and a
+  second one ``foo.Baz``, and dune will compose these properly.
+
+- The ``modules`` field allows one to constrain the set of modules included in
+  the theory, similarly to its OCaml counterpart. Modules are specified in Coq
+  notation. That is to say, ``A/b.v`` is written ``A.b`` in this field.
+
+- If ``package`` is present, Dune will generate install rules for the ``.vo``
+  files of the theory. ``pkg_name`` must be a valid package name.
+  
+  Note that Coq lang 1.0 will use the Coq legacy install setup, where all
+  packages share a common root namespace and install directory,
+  ``lib/coq/user-contrib/<module_prefix>``, as customary in the Make-based Coq
+  package ecosystem.
+  
+  For compatibility, we also install under the ``user-contrib`` prefix the
+  ``.cmxs`` files that appear in ``<ocaml_libraries>``,
+
+- ``<coq_flags>`` will be passed to ``coqc`` as command-line options.
```

Comment:
```suggestion
- ``<coq_flags>`` is passed to ``coqc`` as command-line options.
```

It think it's simpler to use the present tense rather than the future.  I see a lot of future tense here.